SimpleTotp/TotpProvider.cs

```diff
@@ -0,0 +1,187 @@
+﻿// ReSharper disable StringLiteralTypo
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography;
+using System.Text;
+
+[assembly:InternalsVisibleTo("SimpleTotp.Tests")]
+
+namespace SimpleTotp
+{
+    /// <inheritdoc />
+    public class TotpProvider : ITotpProvider
+    {
+        /// <summary>
+        /// Unix Epoch time - in ticks
+        /// </summary>
+        private const long UnixEpochInTicks = 621355968000000000L;
+
+        /// <summary>
+        /// Period for which the same code should be calculated - 30 seconds (in ticks)
+        /// <remarks>
+        /// TOTP specification allows this to be customized; most existing authenticators use a static value of 30 seconds - hence a const
+        /// </remarks>
+        /// </summary>
+        private const long Period = 300000000L;
+
+        /// <summary>
+        /// Number of digits in generated TOTP code
+        /// <remarks>
+        /// TOTP specification allows this to be customized; most existing authenticators use a static value of 6 digits - hence a const
+        /// </remarks>
+        /// </summary>
+        private const int CodeDigits = 6;
+
+        /// <summary>
+        /// Default tolerance for TOTP code validation
+        /// </summary>
+        private static readonly TimeSpan DefaultTolerance = TimeSpan.FromMinutes(1);
+
+        /// <inheritdoc />
+        public RegistrationData GetAuthenticatorRegistrationData(String accountName,
+                                                                 String issuer,
+                                                                 String secretKey = null,
+                                                                 bool prefixAccountNameWithIssuer = true)
+        {
+            ValidationHelper.CheckNotNullOrWhitespace(() => issuer);
```

Comment:
Why use additional helper method? Same method from String is not enough?

Reply:
Please see the commit where this was introduced and check how that line (lines actually) looked like before this helper. Let me know if you think this is an improvement, or unnecessary.